code/lib/core-events/src/errors/server-errors.test.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable local-rules/no-uncategorized-errors */
+import { WebpackCompilationError } from './server-errors';
+
+describe('WebpackCompilationError', () => {
+  it('should correctly handle error with error property', () => {
+    const error = new Error('Custom error message');
+    const data = {
+      error,
+    };
+
+    const webpackError = new WebpackCompilationError(data);
+
+    expect(webpackError.message).toBe(error.message);
+  });
+
+  it('should correctly handle error with error within error', () => {
+    const error = new Error('Custom error message');
+    const data = {
+      error: new Error(),
+    };
+    data.error = error;
+
+    const webpackError = new WebpackCompilationError(data);
+
+    expect(webpackError.message).toBe(error.message);
+  });
+
+  it('should correctly handle error with stats.compilation.errors', () => {
+    const compilationErrors = [new Error('Error 1 message'), new Error('Error 2 message')];
+    const data = new Error() as Error & {
+      error: Error & { stats?: { compilation: { errors: Error[] } } };
+    };
+    data.error = new Error();
+    data.error.stats = {
+      compilation: {
+        errors: compilationErrors,
+      },
+    };
+
+    const webpackError = new WebpackCompilationError(data);
+
+    const expectedErrorMessage = compilationErrors
```

Comment:
Can you hard code this instead? Hard to read what the result is supposed to be in the current form

Reply:
Done!